File: apps/api/app/api/onboarding.py
from fastapi import APIRouter, Depends, HTTPException

from app.core.auth import get_current_manager
from app.onboarding.schemas.onboarding_schemas import AgencyProfileIn
from app.onboarding.services.onboarding_service import (
    get_onboarding_status,
    refresh_onboarding,
    save_agency_profile,
)


router = APIRouter(
    prefix="/api",
    tags=["onboarding"],
)


def _require_org(manager: dict):
    org_id = manager.get("org_id") or manager.get("tenant_org_id")

    if not org_id:
        raise HTTPException(
            status_code=403,
            detail="Missing organization context",
        )

    return org_id


@router.get("/onboarding/status")
def onboarding_status(manager=Depends(get_current_manager)):
    return {
        "status": "ok",
        "onboarding": get_onboarding_status(_require_org(manager)),
    }


@router.post("/onboarding/agency-profile")
def save_profile(
    body: AgencyProfileIn,
    manager=Depends(get_current_manager),
):
    result = save_agency_profile(
        org_id=_require_org(manager),
        user_id=manager["user_id"],
        payload=body.model_dump(),
    )

    return {
        "status": "ok",
        "data": result,
    }


@router.post("/onboarding/refresh")
def refresh(manager=Depends(get_current_manager)):
    return {
        "status": "ok",
        "onboarding": refresh_onboarding(
            org_id=_require_org(manager),
            user_id=manager["user_id"],
        ),
    }

File: apps/web/dashboard/services/onboarding.ts
import { api } from "@/services/api";

export type AgencyProfilePayload = {
  legal_name?: string | null;
  brand_name: string;
  country: string;
  city?: string | null;
  address?: string | null;
  phone?: string | null;
  email?: string | null;
  website?: string | null;
  default_language?: string | null;
  default_currency?: string | null;
  business_type?: string | null;
};

export async function getOnboardingStatus() {
  const response = await api.get("/api/onboarding/status");
  return response.data.onboarding;
}

export async function saveAgencyProfile(data: AgencyProfilePayload) {
  const response = await api.post("/api/onboarding/agency-profile", data);
  return response.data.data;
}

export async function refreshOnboarding() {
  const response = await api.post("/api/onboarding/refresh");
  return response.data.onboarding;
}
